### PR TITLE
Refactor sidebar kpi part 2

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard8**
## Summary

This builds upon #1069. When I tried to apply the new SidebarMetric abstraction to positive tested people I ran into an issue. I was assuming that the bar scale in the sidebar is directly related to the metric value, but this is not always the case. So I made an alt property for specifying the name/property pair for the bar scale when it diverges from the other metric.

Changes & enhancements:

- Strict props typing for metric name and lokalize key.
- Change SidebarBarscale so that it can fetch its own specific data (some prop drilling was required)
- Make data config optional (if there is no bar scale used or other non-default behavior the config can be omitted)
- Add all existing differences to required lists
- I wanted strict typing on the `differenceKey` prop as well but didn't manage yet.
- Data configs are split into 3 separate files

### CODE REVIEW

File: src/metric-config/vr.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { MetricConfig } from './types';
+
+const GREEN = '#69c253';
```
Comment (VWSCoronaDashboard4) [inline src/metric-config/vr.ts:3]: Why not read these from the theme?

File: src/components-styled/sidebar-metric/sidebar-metric.tsx [hunk 2]
```diff
@@ -45,7 +63,11 @@ export function SidebarMetric<T>({
...
 
-  const config = getDataConfig(scope, metricName, metricProperty);
+  const config = getDataConfig(
+    scope,
+    (metricName as unknown) as string,
+    metricProperty
+  );
   const commonText = siteText.common.metricKPI;
 
   const title = get(siteText, [localeTextKey, 'titel_kpi']);
```
Comment (VWSCoronaDashboard4) [inline src/components-styled/sidebar-metric/sidebar-metric.tsx:73]: I cannot comment on the line below: the assert contains a small error in the string provided: % should be a $ to make the interpolation work.

**Review by VWSCoronaDashboard4 — COMMENTED** [review]
Looking good; some suggestions commented.

I hope there will no more/not be much exceptions needed to make it work for all other metrics. Otherwise these components could become harder to maintain than a variety of slightly duplicated components.

File: src/components-styled/sidebar-metric/sidebar-barscale.tsx [hunk 3]
```diff
@@ -1,30 +1,57 @@
...
+}
+
+export function SidebarBarScale<T>({
+  data,
+  scope,
+  metricName,
+  metricProperty,
+  localeTextKey,
+}: SidebarBarScaleProps<T>) {
+  const text = siteText[localeTextKey] as Record<string, string>;
```
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard8) [reply src/components-styled/sidebar-metric/sidebar-barscale.tsx:25]: I'm getting some type issue with the readout value of siteText if I don't convert to record string, so I'm leaving this like it is.

**Review by VWSCoronaDashboard1 — APPROVED** [review]

File: src/metric-config/gm.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+const GREEN = '#69c253';
+const YELLOW = '#D3A500';
+const RED = '#f35065';
+
+/**
+ * Maybe we can make this stricter, but if I use keyof National now, it
+ * complains that I haven't implemented all properties of national. So for now
+ * they are all the same and vague.
+ */
```
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard8) [reply src/metric-config/gm.ts:12]: I think I did try that but still didn't succeed. I will give it another try...

[the remark this answers is not in the document]
Comment (VWSCoronaDashboard8) [reply src/metric-config/gm.ts:12]: Ah this seems to work. I placed Partial in the record first argument before and that doesn't work.

```
type NlMetricKey = MetricKeys<National>;
type NlConfig = Partial<Record<NlMetricKey, Record<string, MetricConfig>>>;
```

**Review by VWSCoronaDashboard10 — COMMENTED** [review]
Approved, next step could be to put stuff like data inside a context so we no longer need all the prop drilling.

**Comment by VWSCoronaDashboard8** [comment]
@VWSCoronaDashboard10 There is not so much prop drilling going on I think. Only max 2 levels I think on the pages & sidebar, and since other props are also passed down so I'm not sure we'll gain much from adding a context.

**Pull request merged**